pkg/apis/core/validation/validation.go

```diff
@@ -5279,3 +5280,34 @@ func ValidateProcMountType(fldPath *field.Path, procMountType core.ProcMountType
 		return field.NotSupported(fldPath, procMountType, []string{string(core.DefaultProcMount), string(core.UnmaskedProcMount)})
 	}
 }
+
+// validateTopologySpreadConstraints validates given TopologySpreadConstraints.
+func validateTopologySpreadConstraints(constraints []core.TopologySpreadConstraint, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	supportedRespModes := sets.NewString(string(core.DoNotSchedule), string(core.ScheduleAnyway))
```

Comment:
We should probably build this constant set once and use it later, instead of rebuilding it every time.

Reply:
+1

Reply:
done.